lib/src/realm_class.dart

```diff
@@ -210,19 +214,33 @@ class Realm implements Finalizable {
   }
 
   /// Deletes a [RealmObject] from this `Realm`.
-  void delete<T extends RealmObject>(T object) => realmCore.deleteRealmObject(object);
+  void delete<T extends RealmObject>(T object) {
+    if (object.realm != this) {
```

Comment:
This will throws the same error if the `object` is unmanaged, which will be the wrong error. We should add a check for unmanaged objects.

Reply:
Good point - added a check + a test.